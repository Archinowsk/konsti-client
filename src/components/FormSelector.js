/* @flow */
import React from 'react'
import { useTranslation } from 'react-i18next'

export type Props = {
  input: Object,
  meta: Object,
  data: Array<any>,
  label: string,
}

const FormSelector = (props: Props) => {
  const {
    input,
    meta: { touched, error },
    data,
    label,
  } = props
  const { t } = useTranslation()

  if (!input) return <p />

  return (
    <React.Fragment>
      <div className='form-selector'>
        <label htmlFor={label}>{label}</label>
        <div className='form-input-wrapper'>
          <select className='form-input' {...input}>
            <option value=''>-</option>
            {data.map(val => (
              <option value={val} key={val}>
                {// First letter to uppercase
                val.charAt(0).toUpperCase() + val.slice(1)}
              </option>
            ))}
          </select>
        </div>
        <div className='help-tip'>
          <p>{t(`tooltip.${input.name}`)}</p>
        </div>
      </div>
      {touched && error && (
        <div className='form-field-error'>
          <span>{error}</span>
        </div>
      )}
    </React.Fragment>
  )
}

export default FormSelector
