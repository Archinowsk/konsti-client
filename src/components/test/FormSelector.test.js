/* @flow */
import React from 'react'
import { shallow } from 'enzyme'
import FormSelector from '../FormSelector'
import type { Props } from '../FormSelector'

describe('FormSelector', () => {
  it('should render correctly', () => {
    const props: Props = {
      input: null,
      meta: { touched: false, error: null },
      data: [],
      label: '',
    }
    const component = shallow(<FormSelector {...props} />)
    expect(component).toMatchSnapshot()
  })
})
