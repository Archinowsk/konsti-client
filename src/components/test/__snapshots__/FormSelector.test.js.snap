// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSelector should render correctly 1`] = `<p />`;
